go-controller/pkg/metrics/master.go

```diff
@@ -380,3 +423,56 @@ func IncrementEgressFirewallCount() {
 func DecrementEgressFirewallCount() {
 	metricEgressFirewallCount.Dec()
 }
+
+func incrementNbCfg(nbClient libovsdbclient.Client) {
+	nbCfgValue += 1
+	if err := libovsdbops.UpdateNBGlobalNbCfg(nbClient, nbCfgValue); err != nil {
+		klog.Errorf("Failed to increment NB_Global nb_cfg value: %v", err)
+		return
+	}
+	metricNbcfgNbdb.Set(float64(nbCfgValue))
+}
+
+// chassisPrivateScraper implements the Collector interface so we can
+// generate chassis metrics on-the-fly per scrape
+type chassisPrivateScraper struct {
+	sbClient libovsdbclient.Client
+}
+
+func (c *chassisPrivateScraper) Describe(ch chan<- *prometheus.Desc) {
+	prometheus.DescribeByCollect(c, ch)
```

Comment:
IIUC this will call `*chassisPrivateScraper.Collect()` which may encounter an error and return no metric hence no descriptor. It would be better to catch the error and call [NewInvalidDesc(err)](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#NewInvalidDesc)

Reply:
Thank you. Corrected this.